### ISSUE TITLE
[Feature]: Zoom to fit in image data view

### ISSUE THREAD

**Issue by JC3** [sqlitebrowser/sqlitebrowser#2955]
### Describe the new feature

I would like a toggle in either the toolbar of the Edit Database Cell pane (in image mode), or in the general preferences dialog, that enables / disables zoom-to-fit when viewing image data.

- When enabled, if the pane is in image mode, the displayed image should be scaled down to entirely fit in the viewer area, maintaining its aspect ratio. 
- When disabled, everything would behave as it does now.

If that approach isn't feasible, my second preference would be to allow zooming in/out with the mousewheel. However, personally I definitely prefer the former more basic approach that just lets me see the whole image at once.

### Does this feature exist in another product or project? Please provide a link

It's a relatively vanilla feature in applications that provide image viewing capabilities. 

### Do you have a screenshot? Please add screenshots to help explain your idea.

Use case is browsing a database where the images a very large compared to the viewer area, for example:

![image](https://user-images.githubusercontent.com/5651184/151200354-c4c581d6-1502-44fa-9750-e880de236b9a.png)

When enabled, the image would be zoomed out, like this:

![image](https://user-images.githubusercontent.com/5651184/151201308-7b1a9138-33ef-4e25-b9aa-5a26b1146fcf.png)

(Not the most interesting example image but you get the idea).

**Comment by JC3** [comment]
I've implemented a proof-of-concept here: https://github.com/JC3/sqlitebrowser/commit/afaa6a76cf371e2f21448fb3d69657f7a71acd48 ([branch](https://github.com/JC3/sqlitebrowser/tree/v3.12.x-jc3-imagezoom)).

If we like it (I mean, *I* like it, obviously 😁), I'll add aspect ratio preservation and make a PR. I branched that off of _v3.12.x_, but I can move to another root; wasn't sure what the policy was here.

See that commit log for details. Here is what it looks like:

https://www.youtube.com/watch?v=jdstYQbl5tQ

![image](https://user-images.githubusercontent.com/5651184/151272233-5cdbcde0-b517-4f4b-b2e4-297dd0b071a9.png)

Lmk.

**Comment by chrisjlocke** [comment]
This is excellent!  I'd use the 'zoom to fit' as default - in your example, you need to know it's a sexy sunset shot, not that it has leaves.  The button would need a tooltip ideally.  A band of bored translators (lol) will then translate that into different languages.

As for the branch, this would get merged into the nightly branch, which then gets cherry-picked into whatever the next release is - major or minor.  So just leave it on your own branch, submit a PR, and someone here will gurgle it down somewhere.  

Thanks again for this - a nice addition!

**Comment by justinclift** [comment]
With the implementation, is that a toggle on/off kind of thing, or does the end user need to click on it for every picture?

If it's a toggle, I wonder if we should have it enabled by default?

**Comment by JC3** [comment]
It's a toggle, same persistence behavior as the word wrap toggle.

I also think it should be on by default. In that case,  instead of calling it "fit image to viewport" and having it on by default, would you rather its behavior be inverted and called "show as actual size" or "1:1" or something (with an icon appropriate for that) and be off by default? 

I like the current form ("fit" + default on) but also idc and its just a minor ux thing so, your call.

**Comment by mgrojo** [comment]
I don't have much time for looking into this, but have you taken into account the latest developments? I'd swear there is already support for zoom-to-fit in the master branch. You can try it in the nightly builds or by building yourself.

**Comment by JC3** [comment]
> I don't have much time for looking into this, but have you taken into account the latest developments? I'd swear there is already support for zoom-to-fit in the master branch. You can try it in the nightly builds or by building yourself.

Nope, I haven't. And I saw a merge conflict into master. Thanks for the heads up I'll take a closer look. The latest dev / candidate for release is maintained on master, yeah?

**Comment by mgrojo** [comment]
> Nope, I haven't. And I saw a potential merge conflict into master that I was wondering about but didn't look at. Thanks for the heads up I'll take a closer look. The latest dev / candidate for release is maintained on master, yeah?

Yes, all the development happens on master, and then we open branches for each release, for fixing any problem for a subsequent bug-fix release.
